snapcraft/internal/errors.py

```diff
@@ -82,6 +78,35 @@ class SnapcraftPartMissingError(SnapcraftError):
     )
 
 
+class SnapcraftLogicError(SnapcraftError):
```

Comment:
`logic` is a bad name. But this was not introduced in this branch and I can't think now of a better name. So just to keep in mind, let's try to find a better one.